evalml/automl/engine/engine_base.py

```diff
@@ -244,6 +285,7 @@ def train_and_score_pipeline(
                 f"\t\t\tFold {i}: {automl_config.objective.name} score: {scores[automl_config.objective.name]:.3f}"
             )
             score = scores[automl_config.objective.name]
+            pipeline_cache[hashes] = cv_pipeline.component_graph.component_instances
```

Comment:
Omega nit-pick since `hashes` is defined within the if/else loop. Can we just call pipeline_cache[hash_dict[i]]? That way, we can remove line 256 too. (we could take this a step further and just pass in get_hashes = i not in hash_dict in the parameters for `train_pipeline` and then set `hash_dict[i] = hashes` when we need to